REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/certificate/validation/CertificateValidator.java

```diff
@@ -75,4 +74,63 @@ private String isValidTagErrorResponse(Tag newTag) {
         }
         return "";
     }
+
+    public String isCreatableGCertificateFieldsWithErrorResponse(GCertificate newCertificate) {
+        StringBuilder errorStringBuilder = new StringBuilder();
+        if (StringUtils.isEmpty(newCertificate.getName()) ||
+                StringUtils.isBlank(newCertificate.getName())) {
+            errorStringBuilder.append("Invalid certificate field name ( name = ")
+                    .append(newCertificate.getName())
+                    .append("). Name cannot be empty!");
+        }
+        for (Variant variant : newCertificate.getVariants()) {
+            if (ObjectUtils.isEmpty(variant.duration()) ||
+                    !(variant.duration() > 0)) {
+                errorStringBuilder.append("Invalid certificate variant field  duration ( duration = ")
+                        .append(variant.duration())
+                        .append("). Duration for new certificate must be positive number!");
+            }
+            for (Variant.Prices prices : variant.prices()) {
+                if (ObjectUtils.isEmpty(prices.centAmount()) ||
+                        prices.centAmount().compareTo(BigDecimal.ZERO) < 0) {
+                    errorStringBuilder.append("Invalid certificate variant field price ( price = ")
+                            .append(prices.centAmount())
+                            .append("). Price for new certificate must be positive or zero!");
+                }
+            }
+        }
+        if (!CollectionUtils.isEmpty(newCertificate.getTags())) {
+            newCertificate.getTags().stream()
+                    .map(this::isValidTagErrorResponse)
+                    .forEach(errorStringBuilder::append);
+        }
+        return errorStringBuilder.toString();
+    }
+
+    public String isUpdatableGCertificateFieldsWithErrorResponse(GCertificate newCertificate) {
+        StringBuilder errorStringBuilder = new StringBuilder();
+        if (!CollectionUtils.isEmpty(newCertificate.getVariants())) {
+            for (Variant variant : newCertificate.getVariants()) {
```

Comment:
Reminder for myself. Check when it will be changed to streams

Reply:
done